editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-execution-scope.ts

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react'
+import type { MapLike } from 'typescript'
+import { isUtopiaJSXComponent, UtopiaJSXComponent } from '../../../core/shared/element-template'
+import { fastForEach } from '../../../core/shared/utils'
+import { ProjectContentTreeRoot } from '../../assets'
+import { importResultFromImports } from '../../editor/npm-dependency/npm-dependency'
+import { TransientFileState } from '../../editor/store/editor-state'
+import {
+  ComponentRendererComponent,
+  createComponentRendererComponent,
+} from './ui-jsx-canvas-component-renderer'
+import { MutableUtopiaContextProps } from './ui-jsx-canvas-contexts'
+import { getTopLevelElements } from './ui-jsx-canvas-top-level-elements'
+
+export function useExecutionScope(
+  filePath: string,
+  customRequire: (importOrigin: string, toImport: string) => any,
+  mutableContextRef: React.MutableRefObject<MutableUtopiaContextProps>, // TODO this is only for the storyboard now
+  projectContents: ProjectContentTreeRoot,
+  openStoryboardFileNameKILLME: string | null,
+  transientFileState: TransientFileState | null,
+): {
+  scope: MapLike<any>
+  topLevelJsxComponents: Map<string, UtopiaJSXComponent>
+  requireResult: MapLike<any>
+} {
+  const { topLevelElements, imports } = getTopLevelElements(
+    filePath,
+    projectContents,
+    openStoryboardFileNameKILLME,
+    transientFileState,
+  )
+  const requireResult: MapLike<any> = importResultFromImports(filePath, imports, customRequire)
```

Comment:
this code is mostly moved verbatim from ui-jsx-canvas, with the exception of the use of `getTopLevelElements` above